test/infrastructure/inmemory/internal/cloud/runtime/cache/client.go

```diff
@@ -117,6 +119,15 @@ func (c *cache) List(resourceGroup string, list client.ObjectList, opts ...clien
 				}
 			}
 
+			// TODO: This only matches the nodeName field for pods. No other fieldSelectors are implemented.
```

Comment:
WDYT about returning an error when we notice another field selector?

Reply:
https://github.com/kubernetes-sigs/cluster-api/pull/8938#discussion_r1252237631